### ISSUE TITLE
Question about the cross attention

### ISSUE THREAD

**Issue by qiguming** [adobe-research/custom-diffusion#82]
if crossattn:
detach = torch.ones_like(key)
detach[:, :1, :] = detach[:, :1, :]0.
key = detachkey + (1-detach)key.detach()
value = detachvalue + (1-detach)*value.detach()

Why stop the gradient of the first key-value pair here?

**Comment by nupurkmr9** [comment]
Hi,

Since the first start of the sentence token is always fixed, I noticed a small improvement when detaching it during the training. I guess this helps in better association between the "V* category" and the target image and thus improved generation on inference time prompt.

Thanks.
